expo/ticketing/models.py

```diff
@@ -146,7 +157,167 @@ class Transaction(models.Model):
     reference = models.CharField(max_length=30)
     payment_source = models.CharField(max_length=30)
     import_date = models.DateTimeField(auto_now=True)
-    
+
     def __unicode__(self):
         return '%s (%s)' % (self.reference, self.purchaser)
-    
+
+
+class CheckListItem(models.Model):
+    '''
+    This is a physical item that we can give away at the registration desk
+    Examples:  a badge, a wristband, a goodie bag, a guidebook, a release
+    form, etc.  It may or may not be labeled for a specific users (for example,
+    a badge has a name on it)
+    '''
+    description = models.CharField(max_length=50, unique=True)
+
+    def __unicode__(self):
+        return unicode(self.description)
+
+
+class EligibilityCondition(models.Model):
+    '''
+    This is the paremt class connecting the conditions under which a
+    CheckListItem can be given to the conditions themselves.
+
+    Conditions are logically additive unless eliminated by exclusions.
+    So if 3 conditions give an item with no exclusion, then the individual
+    gets 3 items.
+
+    TO Discuss (post expo) - consider making this abstract and using content
+    types to link the exclusion foreign key to abstract cases of this class.
+    '''
+    checklistitem = models.ForeignKey(
+        CheckListItem,
+        related_name="%(app_label)s_%(class)s")
+
+    def no_exclusion(self, held_tickets, profile, conference):
+        no_exclusion = True
+        if held_tickets:
+            for exclusion in TicketingExclusion.objects.filter(condition=self):
```

Comment:
Can you use self.ticketing_ticketingexclusion_set.all() instead?  (I think that's the correct unmunged form)
Also, it would be more efficient and readable to convert this loop into an any() expression over a list comprehension.